### DISCUSSION TITLE
Customer impersonation token

### DISCUSSION THREAD

**Discussion by Eddie993** [bigcommerce/catalyst#625] [Q&A] [answered]
Hello,
I have a question about the customer impersonation token, I saw that it's required to use graphql codegen properly, now the question is:
I should generate new customer token for every build (if the previous one is expired) or there is some way to generate a special one that will not expired?

Thank you!

**Comment by christensenep** [comment, marked as answer]
Thanks for the question! No, you do not need to create a new customer impersonation token for every build, you can continue using the same one until it expires. However, once it has expired, yes you will need to generate a new one.

That being said, you can set the expiration time as far into the future as you like (well, as far as Jan 19th, 2038 at least, the maximum unix time value). See https://developer.bigcommerce.com/docs/rest-authentication/tokens/customer-impersonation-token#create-a-token for information on how to create the token, it's the `expires_at` field you'll want to set farther into the future if you want the token to be long-lived.
